src/pages/workspace/WorkspaceMembersPage.tsx

```diff
@@ -1,143 +1,121 @@
-import compose from '@libs/compose';
 import * as DeviceCapabilities from '@libs/DeviceCapabilities';
 import Log from '@libs/Log';
 import Navigation from '@libs/Navigation/Navigation';
 import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
 import * as PolicyUtils from '@libs/PolicyUtils';
 import * as UserUtils from '@libs/UserUtils';
-import personalDetailsPropType from '@pages/personalDetailsPropType';
 import * as Policy from '@userActions/Policy';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {PersonalDetails, PersonalDetailsList, PolicyMember, PolicyMembers} from '@src/types/onyx';
+import type {PendingAction} from '@src/types/onyx/OnyxCommon';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import SearchInputManager from './SearchInputManager';
-import {policyDefaultProps, policyPropTypes} from './withPolicy';
+import type {WithPolicyAndFullscreenLoadingProps} from './withPolicyAndFullscreenLoading';
 import withPolicyAndFullscreenLoading from './withPolicyAndFullscreenLoading';
 
-const propTypes = {
-    /** All personal details asssociated with user */
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
-
-    /** URL Route params */
-    route: PropTypes.shape({
-        /** Params from the URL path */
-        params: PropTypes.shape({
-            /** policyID passed via route: /workspace/:policyID/members */
-            policyID: PropTypes.string,
-        }),
-    }).isRequired,
-
-    /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user accountID */
-        accountID: PropTypes.number,
-    }),
-
-    isLoadingReportData: PropTypes.bool,
-    ...policyPropTypes,
-    ...withLocalizePropTypes,
-    ...windowDimensionsPropTypes,
-    ...withCurrentUserPersonalDetailsPropTypes,
-    network: networkPropTypes.isRequired,
+type WorkspaceMembersPageOnyxProps = {
+    /** Personal details of all users */
+    personalDetails: OnyxEntry<PersonalDetailsList>;
 };
 
-const defaultProps = {
-    personalDetails: {},
-    session: {
-        accountID: 0,
-    },
-    isLoadingReportData: true,
-    ...policyDefaultProps,
-    ...withCurrentUserPersonalDetailsDefaultProps,
-};
+type WorkspaceMembersPageProps = WithPolicyAndFullscreenLoadingProps & WithCurrentUserPersonalDetailsProps & WorkspaceMembersPageOnyxProps;
+
+/**
+ * Inverts an object, equivalent of _.invert
+ */
+function invertObject(object: Record<string, string>): Record<string, string> {
+    const invertedEntries = Object.entries(object).map(([key, value]) => [value, key]);
+    const inverted: Record<string, string> = Object.fromEntries(invertedEntries);
+    return inverted;
+}
 
-function WorkspaceMembersPage(props) {
+type MemberOption = Omit<User, 'accountID'> & {accountID: number};
```

Comment:
Maybe it's possible to update User type to make `accountID` be required?

Reply:
It causes problems in **BaseListItem.tsx** on **line 104**, it expects `item: RadioItem & User` and `accountID` doesn't exist in `RadioItem`